### PR TITLE
Added support for PERF-SAMPLE command (just able to recieve, not func…

### LINKED ISSUE #129: Test packet injector task
Now what we'll want is a means to actually inject the test packets.  To do this, let's create a new `rpc_worker` command, call it `PERF SAMPLE` let's say.  We'll want to parse two arguments from after the command name too, a duration (in seconds) and a sampling rate (in samples per second).

What this command should do is, run for the specified duration, and while it's running, send a `TestPacket` to all queues at the specified sampling rate.  The responses to each `TestPacket` we'll want to aggregate into histograms: each queue and each metric should have its own histogram.  [HdrHistogram](https://docs.rs/hdrhistogram/latest/hdrhistogram/) looks like a decent histogram package.

A good progression of PRs here could be:
1. Send a single test packet to a single queue and send back the values of just that one sample.
2. Send many test packets to a single queue and aggregate the results into one histogram for each metric.
3. Send many test packets to all queues, aggregating the results into a separate histogram for each queue/metric pair.

When the duration ends, the perf sample command should reply with a textual summary of the histograms.

Finally, you'll want to add this command to the debug tool.

Blocked on:
- org-zpr/zpr-core#126

### LINKED ISSUE #131: Change parsing for time for watch command in ph-debug

### PR DESCRIPTION
**Pull request by owenrhysthomas**
…tionality) commands to single word

### CODE REVIEW

**Comment by owenrhysthomas** [comment]
Important to note, in order to test this now, you must setup both a client and server ph, otherwise the command PERF-SAMPLE command will stall because the outbound send packages will stall in the dtls worker. 

Also, all of my queue depth is 0. Unlike the bug where time was zero, this makes sense to me because all of the enqueue_test_packets wait to send the next packet until the previous one was popped off the queue, but my logic may be flawed, so just flagging that.

File: adapter/ph/src/rpc_worker.rs [hunk 1]
```diff
@@ -89,13 +97,85 @@ async fn counters(asm: &Assembly<'_>) -> String {
...
+    let _ = write!(&mut info, "{}", values_from_hist("Inbound Send Depth", " packets", inbound_send_depth).await.as_str());
+    let _ = write!(&mut info, "{}", values_from_hist("Outbound Processor Duration", "ns", outbound_processor_duration).await.as_str());
+    let _ = write!(&mut info, "{}", values_from_hist("Outbound Processor Depth", " packets", outbound_processor_depth).await.as_str());
+    let _ = write!(&mut info, "{}", values_from_hist("Outbound Send Duration", "ns", outbound_send_duration).await.as_str());
+    let _ = write!(&mut info, "{}", values_from_hist("Outbound Send Depth", " packets", outbound_send_depth).await.as_str());
+    
+    info
 }
+
+// TODO not sure if this has to be async
```
Comment (svjaffer) [inline adapter/ph/src/rpc_worker.rs]: I don't think this needs to be async

File: adapter/ph/src/test_packet.rs [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
         let time = Instant::now();
         let t_pkt = TestPacket { time, sender };
 
         (t_pkt, receiver)
     }
 
     pub fn acknowledge(self, queue_depth: usize) {
-        let in_queue = self.time - Instant::now();
+        let curr_time = Instant::now();
+        let in_queue = curr_time.duration_since(self.time);// - Instant::now();
```
Comment (svjaffer) [inline adapter/ph/src/test_packet.rs]: Is this comment dead code or documentation?  If it's the former, remove it and if it's the latter, make it more readable :)

File: adapter/ph-debug/src/main.rs [hunk 3]
```diff
@@ -24,32 +24,27 @@ struct Args {
...
-
-
-    Ok(())
-}
-
-// Determines which command to execute
-fn handle_commands(command: String, time: u64, port: String) -> std::io::Result<()> {
-    println!("{command}, {port}");
-    // fs::remove_file(&port).or_else(|e| if e.kind() == ErrorKind::NotFound { Ok(()) } else { Err(e) }).unwrap();
+    //handle_commands(command, time, port)?;
```
Comment (svjaffer) [inline adapter/ph-debug/src/main.rs]: Dead line

**Review by svjaffer — APPROVED** [review]
Would you mind putting a comment on the PR that shows the output?

**Comment by owenrhysthomas** [comment]
Here's the output from calling the command (I've just removed the words "All values" from the first line under "Message Recieved." Also I fixed the spelling of Received

Message Recieved
All values Inbound Processor Duration values at - 10th Quantile: 31743ns, 25th Quantile: 31743ns,
50th Quantile: 31743ns, 75th Quantile: 31743ns, 90th Quantile: 31743ns

Inbound Processor Depth values at - 10th Quantile: 0 packets, 25th Quantile: 0 packets,
50th Quantile: 0 packets, 75th Quantile: 0 packets, 90th Quantile: 0 packets

Inbound Send Duration values at - 10th Quantile: 13823ns, 25th Quantile: 13823ns,
50th Quantile: 13823ns, 75th Quantile: 13823ns, 90th Quantile: 13823ns

Inbound Send Depth values at - 10th Quantile: 0 packets, 25th Quantile: 0 packets,
50th Quantile: 0 packets, 75th Quantile: 0 packets, 90th Quantile: 0 packets

Outbound Processor Duration values at - 10th Quantile: 10751ns, 25th Quantile: 10751ns,
50th Quantile: 10751ns, 75th Quantile: 10751ns, 90th Quantile: 10751ns

Outbound Processor Depth values at - 10th Quantile: 0 packets, 25th Quantile: 0 packets,
50th Quantile: 0 packets, 75th Quantile: 0 packets, 90th Quantile: 0 packets

Outbound Send Duration values at - 10th Quantile: 47103ns, 25th Quantile: 47103ns,
50th Quantile: 47103ns, 75th Quantile: 47103ns, 90th Quantile: 47103ns

Outbound Send Depth values at - 10th Quantile: 0 packets, 25th Quantile: 0 packets,
50th Quantile: 0 packets, 75th Quantile: 0 packets, 90th Quantile: 0 packets

OK

**Pull request merged**